File: OdonSys-SPA/src/app/core/models/view/doctor-model.ts
import { Country } from "../../enums/country.enum";

export class DoctorModel {
  constructor(
    public id: string,
    public name: string,
    public middleName: string,
    public surname: string,
    public secondSurname: string,
    public document: string,
    public country: Country,
    public email: string,
    public phone: string,
    public userName: string,
    public active: boolean,
    public approved: boolean,
    public roles: string[],
  ) { }
}

File: OdonSysBackEnd/Resource/Contract/Contract.Administration/Users/DoctorModel.cs
using Utilities.Enums;

namespace Contract.Administration.Users;

public record DoctorModel(
    string Id,
    string Name,
    string MiddleName,
    string Surname,
    string SecondSurname,
    string Document,
    Country Country,
    string Email,
    string Phone,
    string UserName,
    bool Active,
    bool Approved,
    IEnumerable<string> Roles
)
{
    public string ExternalUserId { get; set; }
};
